### ISSUE TITLE
Redis omits query offset parameter in key

### ISSUE THREAD

**Issue by krystian50** [cube-js/cube.js#636]
**Describe the bug**
I'm querying sorted data. Changing offset seems to be ignored by the query.

**To Reproduce**
1. Use query eg: 
`{"dimensions":["Refunds.refundNumber"],"limit":10,"offset":0,"order":{"Refunds.refundNumber":"asc"}}`

this query returns example data:
```
{
    "query": {
        "dimensions": [
            "Refunds.refundNumber"
        ],
        "limit": 10,
        "offset": 0,
        "order": [
            {
                "id": "Refunds.refundNumber",
                "desc": false
            }
        ],
        "rowLimit": 10,
        "timezone": "UTC",
        "filters": [],
        "timeDimensions": []
    },
    "data": [
        {
            "Refunds.refundNumber": "691695"
        },
        {
            "Refunds.refundNumber": "691696"
        },
        {
            "Refunds.refundNumber": "691697"
        },
...
```
2. Change offset in query
`{"dimensions":["Refunds.refundNumber"],"limit":10,"offset":10,"order":{"Refunds.refundNumber":"asc"}}`
this query returns:
```
{
    "query": {
        "dimensions": [
            "Refunds.refundNumber"
        ],
        "limit": 10,
        "offset": 10,
        "order": [
            {
                "id": "Refunds.refundNumber",
                "desc": false
            }
        ],
        "rowLimit": 10,
        "timezone": "UTC",
        "filters": [],
        "timeDimensions": []
    },
    "data": [
        {
            "Refunds.refundNumber": "691695"
        },
        {
            "Refunds.refundNumber": "691696"
        },
        {
            "Refunds.refundNumber": "691697"
        },
...
```
**Expected behavior**
The query should return data window moved by 10

**Screenshots**

**Version:**
[0.19.16] - server
[0.19.15] - postgres-driver

**Comment by krystian50** [comment]
I just found out my issue is caused by `offset` parameter being ignored while setting up cache key in redis. Basically first request with `offset: 0` is cached by redis. The `query.offset` is changed in result, but data keep cached fo `offset: 100`.

**Comment by krystian50** [comment]
I found the only way to make it works is by completely removing cache for compiler: 
```
class CompilerCache extends QueryCache {
  constructor({ maxQueryCacheSize, maxQueryCacheAge }) {
    super();
    console.log(maxQueryCacheSize, maxQueryCacheAge)
    this.queryCache = new LRUCache({
      max: maxQueryCacheSize || 1,
      maxAge: (maxQueryCacheAge * 1000) || 1,
      updateAgeOnGet: true
    });
  }
```

**Comment by paveltiunov** [comment]
@krystian50 Hey Krystian! Thanks for posting this one! Yep. Seems like a bug. Should be fixed in 0.19.18. Could you please check?
